tests/AerospikeCacheTest.php

```diff
@@ -0,0 +1,150 @@
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock->method('put')
+                ->willReturn($statusCode);
+
+        $aerospikeMock->method('get')
+                ->willReturn(\Aerospike::OK);
+
+        $aerospikeKey = ['ns' => 'test', 'set' => 'cache', 'key' => 'foo'];
+        $aerospikeMock->method('initKey')
+                ->willReturn($aerospikeKey);
+
+        $cacheItem = $aerospikeCache->getItem('foo');
+        $result = $aerospikeCache->save($cacheItem);
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    public function doSaveProvider(): array
+    {
+        return [
+            [\Aerospike::OK, true],
+            [\Aerospike::ERR_CLIENT, false],
+        ];
+    }
+
+    /**
+     * @dataProvider doClearWithEmptyNamespaceProvider
+     */
+    public function testDoClearWithEmptyNamespace(int $statusCode, bool $expectedResult): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock->method('truncate')->willReturn($statusCode);
+
+        $result = $aerospikeCache->clear();
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    public function doClearWithEmptyNamespaceProvider(): array
+    {
+        return [[\Aerospike::OK, true] , [\Aerospike::ERR_CLIENT, false]];
+    }
+
+    /**
+     * @dataProvider doClearNamespaceProvider
+     * @param mixed $expectedValue
+     * @param mixed $statusCodeForRemove
+     * @param mixed $statusCodeForScan
+     */
+    public function testDoClearNamespace(int $statusCodeForRemove, int $statusCodeForScan, bool $expectedValue): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
+
+        $aerospikeMock->method('remove')->willReturn($statusCodeForRemove);
+
+        $aerospikeMock->method('scan')->willReturnCallback(function ($namespace, $set, $callback) use ($statusCodeForScan) {
+            call_user_func($callback, ['key' => ['key' => 'testNamespace::test']]);
+
+            return $statusCodeForScan;
+        });
+
+        $clearResult = $aerospikeCache->clear('testNamespace');
+
+        $this->assertEquals($expectedValue, $clearResult);
+    }
+
+    public function doClearNamespaceProvider(): array
+    {
+        return [
+            [\Aerospike::OK, \Aerospike::OK, true],
+            [\Aerospike::OK, \Aerospike::ERR_CLIENT, false],
+            [\Aerospike::ERR_CLIENT, \Aerospike::ERR_CLIENT, false],
+            [\Aerospike::ERR_CLIENT, \Aerospike::OK, false],
+        ];
+    }
+
+    /**
+     * @dataProvider doDeleteProvider
+     */
+    public function testDoDelete(int $aerospikeStatus, bool $expectedValue): void
```

Comment:
```suggestion
    public function testShouldDeleteItem(int $aerospikeStatus, bool $expectedValue): void
```

Reply:
changed. Also changed test names of few other methods in this fashion